tests/models/superglue/test_modeling_superglue.py

```diff
@@ -0,0 +1,337 @@
+            "keypoint_decoder_dim": 65,
+            "descriptor_decoder_dim": 64,
+            "keypoint_threshold": 0.005,
+            "max_keypoints": 256,
+            "nms_radius": 4,
+            "border_removal_distance": 4,
+        },
+        descriptor_dim: int = 64,
+        keypoint_encoder_sizes: List[int] = [32, 64],
+        gnn_layers_types: List[str] = ["self", "cross"] * 2,
+        num_heads: int = 4,
+        sinkhorn_iterations: int = 100,
+        matching_threshold: float = 0.2,
+    ):
+        self.parent = parent
+        self.batch_size = batch_size
+        self.image_width = image_width
+        self.image_height = image_height
+
+        self.keypoint_detector_config = keypoint_detector_config
+        self.descriptor_dim = descriptor_dim
+        self.keypoint_encoder_sizes = keypoint_encoder_sizes
+        self.gnn_layers_types = gnn_layers_types
+        self.num_heads = num_heads
+        self.sinkhorn_iterations = sinkhorn_iterations
+        self.matching_threshold = matching_threshold
+
+    def prepare_config_and_inputs(self):
+        # SuperGlue expects a grayscale image as input
+        pixel_values = floats_tensor([self.batch_size, 2, 3, self.image_height, self.image_width])
+        config = self.get_config()
+        return config, pixel_values
+
+    def get_config(self):
+        return SuperGlueConfig(
+            keypoint_detector_config=self.keypoint_detector_config,
+            descriptor_dim=self.descriptor_dim,
+            keypoint_encoder_sizes=self.keypoint_encoder_sizes,
+            gnn_layers_types=self.gnn_layers_types,
+            num_heads=self.num_heads,
+            sinkhorn_iterations=self.sinkhorn_iterations,
+            matching_threshold=self.matching_threshold,
+        )
+
+    def create_and_check_model(self, config, pixel_values):
+        model = SuperGlueForKeypointMatching(config=config)
+        model.to(torch_device)
+        model.eval()
+        result = model(pixel_values)
+        maximum_num_matches = result.mask.shape[-1]
+        self.parent.assertEqual(
+            result.keypoints.shape,
+            (self.batch_size, 2, maximum_num_matches, 2),
+        )
+        self.parent.assertEqual(
+            result.matches.shape,
+            (self.batch_size, 2, maximum_num_matches),
+        )
+        self.parent.assertEqual(
+            result.matching_scores.shape,
+            (self.batch_size, 2, maximum_num_matches),
+        )
+
+    def prepare_config_and_inputs_for_common(self):
+        config_and_inputs = self.prepare_config_and_inputs()
+        config, pixel_values = config_and_inputs
+        inputs_dict = {"pixel_values": pixel_values}
+        return config, inputs_dict
+
+
+@require_torch
+class SuperGlueModelTest(ModelTesterMixin, unittest.TestCase):
+    all_model_classes = (SuperGlueForKeypointMatching,) if is_torch_available() else ()
+    all_generative_model_classes = () if is_torch_available() else ()
+
+    fx_compatible = False
```

Comment:
At first I thought this would be needed in order to pass the tests, turns out not. But out of curiosity I tried and SuperGlue is indeed not compatible, since some operations depend on the input data. I removed this line